src/processor/operator/install_extension.cpp

```diff
@@ -0,0 +1,64 @@
+#include "processor/operator/install_extension.h"
+
+#include "common/exception/io.h"
+#include "common/file_system/virtual_file_system.h"
+#include "common/string_format.h"
+#include "extension/extension.h"
+#include "httplib.h"
+#include "main/database.h"
+
+namespace kuzu {
+namespace processor {
+
+using namespace kuzu::common;
+using namespace kuzu::extension;
+
+bool InstallExtension::getNextTuplesInternal(ExecutionContext* context) {
+    if (!hasExecuted) {
+        hasExecuted = true;
+        installExtension(context->vfs, context->database);
+    }
+    return false;
+}
+
+std::string InstallExtension::tryDownloadExtension() {
+    auto extensionRepoInfo = ExtensionUtils::getExtensionRepoInfo(name);
+    httplib::Client cli(extensionRepoInfo.hostURL.c_str());
+    httplib::Headers headers = {
+        {"User-Agent", common::stringFormat("kuzu/v{}", KUZU_RELEASE_VERSION)}};
+    auto res = cli.Get(extensionRepoInfo.hostPath.c_str(), headers);
+    if (!res || res->status != 200) {
+        if (res.error() == httplib::Error::Success) {
+            // LCOV_EXCL_START
+            throw IOException(common::stringFormat(
```

Comment:
How can we trigger this exception? Can we set up a test case against this?

Reply:
The server maybe busy or we used invalid header, the server may return a code rather than 200 with an error message.